controllers/scopetemplate_controller_test.go

```diff
@@ -0,0 +1,548 @@
+					return err
+				}
+
+				if len(clusterRoleList.Items) != 0 {
+					return fmt.Errorf("Expected 0 clusterRoles, found %d", len(clusterRoleList.Items))
+				}
+
+				return nil
+
+			}, timeout, interval).Should(BeNil())
+		})
+
+		When("a scopeInstance is created that references the scopeTemplate with a single namespace", func() {
+			BeforeEach(func() {
+				scopeInstance = &operatorsv1.ScopeInstance{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "ScopeInstance",
+						APIVersion: "operators.io.operator-framework/v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "scopeinstance-sample",
+					},
+					Spec: operatorsv1.ScopeInstanceSpec{
+						ScopeTemplateName: scopeTemplate.GetName(),
+						Namespaces:        []string{namespace.GetName()},
+					},
+				}
+				err := k8sClient.Create(ctx, scopeInstance)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			AfterEach(func() {
+				Expect(k8sClient.Delete(ctx, scopeInstance)).NotTo(HaveOccurred())
+			})
+
+			It("should create the expected clusterRole", func() {
+				clusterRoleList := &rbacv1.ClusterRoleList{}
+
+				Eventually(func() error {
+					err := k8sClient.List(ctx, clusterRoleList, client.MatchingLabels{
+						scopeTemplateUIDKey:    string(scopeTemplate.GetUID()),
+						clusterRoleGenerateKey: scopeTemplate.Spec.ClusterRoles[0].GenerateName,
+					})
+
+					if err != nil {
+						return err
+					}
+
+					if len(clusterRoleList.Items) != 1 {
+						return fmt.Errorf("expected 1 clusterRole, found %d", len(clusterRoleList.Items))
+					}
+
+					return nil
+
+				}, timeout, interval).Should(BeNil())
+
+				existingRole := &clusterRoleList.Items[0]
+
+				Expect(len(existingRole.OwnerReferences)).Should(Equal(1))
+				Expect(existingRole.OwnerReferences).Should(ContainElement(metav1.OwnerReference{
+					APIVersion: "operators.io.operator-framework/v1",
+					Kind:       "ScopeTemplate",
+					Name:       scopeTemplate.GetName(),
+					UID:        scopeTemplate.GetUID(),
+				}))
+				Expect(existingRole.Rules).Should(Equal([]rbacv1.PolicyRule{
+					{
+						Verbs:     []string{"get", "watch", "list"},
+						APIGroups: []string{""},
+						Resources: []string{"secrets"},
+					},
+				}))
+			})
+
+			It("creates the expected RoleBinding for test namespace", func() {
+				roleBindingList := &rbacv1.RoleBindingList{}
+				Eventually(func() error {
+					err := k8sClient.List(ctx, roleBindingList, &client.ListOptions{
+						Namespace: namespace.GetName(),
+					},
+						client.MatchingLabels{
+							scopeInstanceUIDKey:           string(scopeInstance.GetUID()),
+							scopeTemplateUIDKey:           string(scopeTemplate.GetUID()),
+							clusterRoleBindingGenerateKey: "test",
+						})
+					if err != nil {
+						return err
+					}
+
+					if len(roleBindingList.Items) != 1 {
+						return fmt.Errorf("Expected 1 roleBinding, found %d", len(roleBindingList.Items))
+					}
+
+					return nil
+				}, timeout, interval).Should(BeNil())
```

Comment:
Thoughts on moving this to its own function? Parameters should be the namespace, matchingLabels, and number of roleBindings.

Reply:
Looks perfect.

Reply:
Updated it.